.github/workflows/continuous-deploy.yml

```diff
@@ -239,10 +246,20 @@ jobs:
           name: coverage-files-${{ matrix.module.id }}
           path: coverage-files/.coverage.${{ matrix.module.id }}
 
+  status-check:
+    name: Status Check
+    runs-on: ubuntu-latest
+    needs: test
+    if: always()
+    steps:
+      - name: Check if any test failed
+        run: exit 1
+        if: needs.test.outputs.test_success == 'false'
+
   combine-coverage:
     name: Coverage
-    needs: test
     runs-on: ubuntu-latest
+    needs: status-check
```

Comment:
If any job in the test matrix fails, then the `combine-coverage` job would have been skipped.

Wouldn't it be easier to add `Coverage` as a required check to PR's instead of adding `Status Check` and the above `if else`?